### PR TITLE
Improve consume/produce examples

### PR DESCRIPTION
**Pull request by sylvainb**
Use asyncio.gather instead of asyncio.ensure_future in the first example (more understandable). Add  comments to the second example.

### CODE REVIEW

File: examples/producer_consumer.py [hunk 1]
```diff
@@ -32,6 +32,7 @@
 
 loop = asyncio.get_event_loop()
 queue = asyncio.Queue(loop=loop)
-asyncio.ensure_future(produce(queue, 10), loop=loop)
```
Comment (vstinner) [inline examples/producer_consumer.py:35]: IMO it is deliberate to not wait for the completion of the producer.

Comment (vxgmichel) [reply examples/producer_consumer.py:35]: Yes, the idea here is to run the producer in a background task while waiting for the consumer to complete. However, I'm not opposed to this change if you think it makes the example more understandable. As an alternative, it is possible to make the producer task more explicit:

```
loop = asyncio.get_event_loop()
queue = asyncio.Queue(loop=loop)
producer_task = loop.create_task(produce(queue, 10))
loop.run_until_complete(consume(queue))
assert producer_task.done()
loop.close()
```

In any case, it's probably good to [discourage the fire and forget approach](https://github.com/python/asyncio/issues/399) for the background task.

Comment (ludovic-gasc) [reply examples/producer_consumer.py:35]: Yesterday, during AsyncIO workshop at PyCON-FR, it was hard for two attendees to understand why we have ensure_future for one coroutine, and not for the other.

I've proposed to change the example as proposed by @sylvainb, it seems more understandable by available beginners.
I've also suggested to propose a pull request for that.

Comment (vxgmichel) [reply examples/producer_consumer.py:35]: Then it's fine with me, I'm merging it ;-)

**Comment by vxgmichel** [comment]
Thanks!

**Pull request merged**